src/verso/Verso/Doc/Elab/Monad.lean

```diff
@@ -126,12 +126,22 @@ structure DocElabContext where
   genreSyntax : Syntax
   genre : Expr
 
+
+  /-- Whether references to undefined (not-yet-defined) footnotes and links are permitted. -/
   refsAllowed : RefsAllowed
+
+  /--
+  The docReconstructionPlaceholder provides a free variable during Verso document elaboration. This
+  syntax object cannot be successfully elaborated to a term until closed as a function
+  {lit}`` `(fun $docReconstructionPlaceholder => $termContainingFreeVariable)``.
+  -/
+  docReconstructionPlaceholder : Option Ident
 deriving Inhabited
 
+
 def DocElabContext.fromGenreTerm (genreSyntax : Term) : TermElabM DocElabContext := do
   let genre ← Term.elabTerm genreSyntax (some (.const ``Doc.Genre []))
-  return DocElabContext.mk genreSyntax genre .always
+  return DocElabContext.mk genreSyntax genre .always (.some <| mkIdent (← mkFreshUserName `docRecon))
```

Comment:
I wrote this on the other one, but I figure it makes sense here too.

In my head, "recon" bascially always abbreviates "reconnaissance" rather than "reconstruction", which lead to me always needing to do a double-take on this abbreviation.